tests/prepare_test_data.py

```diff
@@ -0,0 +1,179 @@
+        'skip-dirs': ['Godeps'],
+    },
+    {
+        'name': 'github.com/cznic/ql',
+        'clone-url': 'https://github.com/cznic/ql',
+        'commits': ['77545ff365ff426dfab06c46d70fa77600cd55af'],
+        'skip-dirs': [],
+    },
+    {
+        'name': 'github.com/google/cayley',
+        'clone-url': 'https://github.com/google/cayley',
+        'commits': ['792fe5d42c4dd4d967c3db2e30bd380fcdfb3151'],
+        'skip-dirs': ['Godeps'],
+    },
+    {
+        'name': 'github.com/influxdata/influxdb',
+        'clone-url': 'https://github.com/influxdata/influxdb',
+        'commits': [
+            'f326c6c0e4349a6f9bca056571b01ded81a2887c',
+            '688bc7a2f1877b524a3c915ae1ff7981267fcb49',
+            'd771eb72f708c523bb2bec35a97f6bb83ae4ef1d',
+        ],
+        'skip-dirs': [],
+    },
+    {
+        'name': 'github.com/google/martian',
+        'clone-url': 'https://github.com/google/martian',
+        'commits': [
+            '4d4cb10a32e7f1237228d58ecf95c571b0947a11',
+            '1afb734cc2eb3a5dab708a901ee07106a94a9186',
+        ],
+        'skip-dirs': [],
+    },
+]
+
+
+def get_logger():
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    formatter = logging.Formatter('%(levelname)s: %(message)s')
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    return logger
+
+
+def prepare(destdir, projects):
+    logger = get_logger()
+    for project in projects:
+        logger.info('Preparing project "{}"'.format(project['name']))
+        projectdir = os.path.join(destdir, project['name'])
+        try:
+            os.makedirs(projectdir)
+        except OSError:
+            logger.error('Cannot create directory "{}"'.format(projectdir))
+            continue
+        repodir = os.path.join(projectdir, 'repo')
+        try:
+            repo = git.Repo.clone_from(project['clone-url'], repodir)
+        except git.exc.GitCommandError:
+            logger.error('Cannot clone git repository "{}" to "{}"'.format(
+                project['clone-url'], repodir))
+            continue
+        try:
+            for commit in project['commits']:
+                logger.info('Processing commit "{}"'.format(commit))
+                commitdir = os.path.join(projectdir, commit)
+                data = {
+                    'source_code_directory': repodir,
+                    'directories_to_skip': project['skip-dirs'],
+                    'project': project['name'],
+                    'commit': commit,
+                    'ipprefix': project['name'],
+                }
+                datafile = '{}.json'.format(commitdir)
+                try:
+                    with open(datafile, 'wb') as f:
+                        json.dump(data, f)
+                except IOError:
+                    logger.error('Cannot open file "{}"'.format(datafile))
+                    continue
+                archive = '{}.tar.gz'.format(commitdir)
```

Comment:
Two projects can have the same commit (no likely but they can). So create new class into (/system/helpers/) that will be resposible for generating name or a tarball from (project, commit) pair. Then use it to generate the name instead of using commitdir.